### PR TITLE
feat: Decision instance get by key - API implementaton

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description

Implement Decision instance get by key API
`evaluatedInputs` and `matchedRules` are returned in the response, in opposite to the search service for which these fields are omitted for performance reasons

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

relates to [#22656](https://github.com/camunda/camunda/issues/22656)

### CODE REVIEW

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java [hunk 1]
```diff
@@ -334,6 +337,54 @@ private static DecisionInstanceItem toDecisionInstance(final DecisionInstanceEnt
...
+        .map(
+            input ->
+                new EvaluatedDecisionInputItem()
+                    .inputId(input.id())
+                    .inputName(input.name())
+                    .inputValue(input.value()))
+        .toList();
+  }
+
+  private static List<MatchedDecisionRuleItem> toMatchedRules(
```
Comment (houssain-barouni) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java:360]: @ralfpuchert the `MatchedDecisionRules` are flattened by Operate importer to a List of EvaluatedOutputs, here I try to regroup them by ruleId/ruleIndex, can you confirm this logic is correct?

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -1382,6 +1382,59 @@ paths:
               schema:
                 $ref: "#/components/schemas/ProblemDetail"
 
+  /decision-instances/{decisionInstanceKey}:
+    get:
+      tags:
+        - Decision instance
+      summary: Get decision instance by key (alpha)
```
Comment (ralfpuchert) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1389]: Do we need a period at the end here?

**Review by ralfpuchert — APPROVED** [review]

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml (hunk 2 above)
Comment (houssain-barouni) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1389]: No, all summaries in the file do not end with a period

**Review by pepopowitz — APPROVED** [review]
👍 Overall the spec looks good. My feedback is non-blocking, but I do have some recommendations and conversation-starters.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 3]
```diff
@@ -3841,6 +3894,20 @@ components:
...
+          description: |
+            The evaluated inputs of the decision instance.
+            This is returned only by the get decision instance by key endpoint.
+        matchedRules:
+          type: array
+          items:
+            $ref: "#/components/schemas/MatchedDecisionRuleItem"
+          description: |
+            The matched rules of the decision instance.
+            This is returned only by the get decision instance by key endpoint.
```
[the remark this answers is not in the document]
Comment (houssain-barouni) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3910]: that makes sense, I will do that

**Review by christinaausley — APPROVED** [review]

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 4]
```diff
@@ -1382,6 +1382,59 @@ paths:
...
+        "200":
+          description: >
+            The decision instance is successfully returned.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/DecisionInstanceItem"
+        "400":
+          description: >
+            The decision instance Get failed.
```
[the remark this answers is not in the document]
Comment (christinaausley) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml]: 100% agreed on this.

**Pull request merged**